plugins/webp-uploads/hooks.php

```diff
@@ -52,21 +52,30 @@
  * } An array with the updated structure for the metadata before is stored in the database.
  */
 function webp_uploads_create_sources_property( array $metadata, int $attachment_id ): array {
-	// This should take place only on the JPEG image.
+	$file = get_attached_file( $attachment_id, true );
+	// File does not exist.
+	if ( false === $file || ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	/*
+	 * We need to get the MIME type ideally from the file, as WordPress Core may have already altered it.
+	 * The post MIME type is typically not updated during that process.
+	 */
+	$filetype = wp_check_filetype( $file );
+	if ( isset( $filetype['type'] ) ) {
+		$mime_type = $filetype['type'];
+	} else {
+		$mime_type = get_post_mime_type( $attachment_id );
+	}
+
 	$valid_mime_transforms = webp_uploads_get_upload_image_mime_transforms();
 
 	// Not a supported mime type to create the sources property.
-	$mime_type = get_post_mime_type( $attachment_id );
 	if ( ! is_string( $mime_type ) || ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
```

Comment:
Nit-pick: The `! is_string( $mime_type )` early return check could happen before the call to `webp_uploads_get_upload_image_mime_transforms()` to avoid that function call when not relevant.

Reply:
Suggested change above.

```suggestion
	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
```